src/implementations/title-escrow/helpers.ts

```diff
@@ -0,0 +1,71 @@
+import { TitleEscrowFactory, TradeTrustErc721Factory } from "@govtechsg/token-registry";
+import { Wallet } from "ethers";
+import signale from "signale";
+import { ConnectedSigner } from "../utils/wallet";
+
+interface ConnectToTitleEscrowArgs {
+  tokenId: string;
+  address: string;
+  wallet: Wallet | ConnectedSigner;
+}
+
+type TitleEscrowInstanceType = ReturnType<typeof TitleEscrowFactory.connect>;
+
+export const connectToTitleEscrow = async ({
+  tokenId,
+  address,
+  wallet,
+}: ConnectToTitleEscrowArgs): Promise<TitleEscrowInstanceType> => {
+  const tokenRegistry = await TradeTrustErc721Factory.connect(address, wallet);
+  const titleEscrowAddress = await tokenRegistry.ownerOf(tokenId);
+  const titleEscrow = await TitleEscrowFactory.connect(titleEscrowAddress, wallet);
+  return titleEscrow;
+};
+
+interface validateEndorseOwnerArgs {
+  newHolder: string;
+  newOwner: string;
+  titleEscrow: TitleEscrowInstanceType;
+}
+export const validateEndorseOwner = async ({
+  newHolder,
+  newOwner,
+  titleEscrow,
+}: validateEndorseOwnerArgs): Promise<void> => {
+  const beneficiary = await titleEscrow.beneficiary();
+  const holder = await titleEscrow.holder();
+  if (newOwner === beneficiary && newHolder === holder) {
+    const error = "new owner and new holder addresses are the same as the current owner and holder addresses";
+    signale.error(error);
+    throw new Error(error);
+  }
+};
+
+interface validateNominateOwnerArgs {
+  newOwner: string;
+  titleEscrow: TitleEscrowInstanceType;
+}
+export const validateNominateOwner = async ({ newOwner, titleEscrow }: validateNominateOwnerArgs): Promise<void> => {
+  const beneficiary = await titleEscrow.beneficiary();
+  if (newOwner === beneficiary) {
+    const error = "new owner address is the same as the current owner address";
+    signale.error(error);
+    throw new Error(error);
+  }
+};
+
+interface validateEndorseTransferOwnerArgs {
+  approvedOwner: string | undefined;
+  approvedHolder: string | undefined;
+}
+const GENESIS_ADDRESS = "0x0000000000000000000000000000000000000000";
```

Comment:
There isn't any different. just FYI that we could use etherjs constants address zero instead of declaring own one
https://docs.ethers.io/v5/api/utils/constants/#constants-AddressZero

Reply:
nice 👌